### ISSUE TITLE
doesn't work?

### ISSUE THREAD

**Issue by mutsurhea** [cristan2/steam-wishlist-demos#1]
i found the site
https://steam-wishlist-demos.fly.dev/
i found my ID on my account details
i put it into the box and clicked submit
and got zero feedback anything was happening

several minutes went by with nothing

i tabbed over
logged into github
opened my email for a 2fa code
so that i could make a "the thing is broken" thread

and when i finally get to typing
NOW i see the output

had i closed the tab after waiting a reasonable amount of time
id have never known it to be working

i believe the unexpected wait may be due to having a large wishlist
over 1000

site needs like a spinny thingy, and a "downloading wishlist..." "looking for demos...." feedback thingy

**Comment by mutsurhea** [comment]
ok now im going through the output
theres far more than i was expecting
more than 1 full page of 200 entries are tagged demo

except, not all of these results have demos

of the first 6 ive opened, 2 have no demo
https://store.steampowered.com/app/333250/Forward_to_the_Sky/
https://store.steampowered.com/app/1175360/Doomsday_Hunters/

more
https://store.steampowered.com/app/1421490/Rise_of_the_Triad_Ludicrous_Edition/
https://store.steampowered.com/app/1644940/Loddlenaut/

does something need updating?

**Comment by cristan2** [comment]
You're right, there needs to be a loading indicator (as a workaround, you can rely on the loading indicator in the browser's tab title). I have plans for several improvements, but right now it's just a quick prototype thrown together to solve a problem I had. I might improve some things when I have the time.

As for the links you posted, I'm seeing demos for all of them. Maybe they're just not supported in your region?
![image](https://github.com/cristan2/steam-wishlist-demos/assets/13849765/26e684a9-aa6f-4b4c-8fd7-46bca7d0d44d)

![image](https://github.com/cristan2/steam-wishlist-demos/assets/13849765/eae7065c-55b0-4f5d-8fd0-bfcd520b793a)

Does the Install Demo link in the results page do anything for you? Normally it should trigger the installation of the demo.

**Comment by mutsurhea** [comment]
wtf, demos can be regional!?
why on earth
ive asked some of the devs on twitter 
[ [Doomsday Hunters](https://twitter.com/mtsRhea/status/1734216007797846483), [Rise of the Triad](https://twitter.com/mtsRhea/status/1734217515805409674), [Loddlenaut](https://twitter.com/mtsRhea/status/1734217684659621950) ]

what region are you?

> loading indicator in the browser's tab

i hadn't even realised i had one, the visual contrast is so low with my dark theme
![image](https://github.com/cristan2/steam-wishlist-demos/assets/77935124/7085f339-3356-4954-9515-0edf37113179)

> the Install Demo link

i CAN click it!
it works
thats good at least

oh wait a minute, i found the demo link on the steam page
its not in a green box like normal
its off to the side, stealth demo
<img width="807" alt="image" src="https://github.com/cristan2/steam-wishlist-demos/assets/77935124/29e82e8c-c95e-4653-a34e-2600423ef4a2">

**Comment by mutsurhea** [comment]
but wait theres more!

i checked the 4 examples i posted before
all have demos in the sidebar instead of the green centre

ive found an example which has NEITHER

https://store.steampowered.com/app/21980/Call_of_Juarez_Bound_in_Blood/

 but the link on the tool DOES work!

**Comment by cristan2** [comment]
Yeah, the developers have some flexibility in how they present their demos, including the option to remove the link (but it can be found using the API, if the demo still exists). I hadn't noticed the link in the sidebar either.

**Comment by mutsurhea** [comment]
interesting
ty for your tool!

also, one of the devs on [twitter responded](https://twitter.com/loddlenaut/status/1734273642815533253)

>  We moved the demo link to the sidebar once the game launched to avoid confusion between the demo button and the main "Buy Loddlenaut" button. The sidebar button is actually the default demo setting on Steam, weirdly enough!

**Comment by mutsurhea** [comment]
other feature requests

1] hide non-demo games by default in a toggle box

2] order of games should be in wishlist order by default
maybe this should be a bug report
i have 500 games which have No. 0, so i have to go to page 6 to see the priority order

![image](https://github.com/cristan2/steam-wishlist-demos/assets/77935124/298c4bc1-ecd4-4acc-b03f-050ffeb411f6)
![image](https://github.com/cristan2/steam-wishlist-demos/assets/77935124/29b90a74-cf24-4024-966a-6c19ba6d3a19)
![image](https://github.com/cristan2/steam-wishlist-demos/assets/77935124/0c9911ec-e99b-4a2c-bce3-210401e67e12)

 but then that's not super useful as all my highest priority games are AAA,
and those generally dont have demos

ordering by the final column [Demo?] destroys the priority order

im not sure have the 2ndry ordering is when you do this, its not alphabetical

<img width="716" alt="image" src="https://github.com/cristan2/steam-wishlist-demos/assets/77935124/e7218af4-c1d0-491c-9f9a-756196303db1">

**Comment by cristan2** [comment]
1 - That's a useful feature, indeed. There needs to be a filtering toggle for sure, I'll try to do it as soon as I have time. For now, you can get the same result by using the search box to search for "demo":

before:
![image](https://github.com/cristan2/steam-wishlist-demos/assets/13849765/b1d44f1a-e278-4320-b3d6-6209524e87e8)

after:
![image](https://github.com/cristan2/steam-wishlist-demos/assets/13849765/f2a8447a-8592-4907-91c1-4979c477415e)

2 - The problem now is that they *are* in order. Games without a specified priority have priority 0, which will always come before 1 in any ordering. So there needs to be some special treatment for games with priority 0 so that they can be displayed at the end of the ordered list. I'll add this to the todo list.

Thanks for the comments. Glad you're finding some use for the tool.

**Comment by mutsurhea** [comment]
thx

i wanted to make this clear about my suggestions - 
if you cba or dont wanna put more time into the tool, then dont
dont do work because i asked for it
the tool is already doing its job, finding the demos. im extremely grateful for that

im just sharing my user feedback if that interests you

with that said

3] i think many would like a batch download button
1 button which clicks all the download demos
[i wouldn't use it, because adding 200 demos all at once, that's too many. im doing it slowly

**Comment by cristan2** [comment]
No worries, I'll be working on it when I have the time. Thanks for your suggestions, they're welcome.

As for #3, I'm not sure it can be done, it depends on the Steam API. I'll look into it.

**Comment by mutsurhea** [comment]
found something unusual i thought you might find mildly interesting

when you go to Indivisible and play the demo
https://store.steampowered.com/app/421170/Indivisible/
and then go to your Recently Played Games
the playtime is logged on a separate database entry Indivisible Prototype Demo
https://store.steampowered.com/app/421830/Indivisible_Prototype_Demo/

> Indivisible Prototype Demo is already in your Steam library

however this entry does not show up in my Library

i think this technique would be useful for Outpath
it has a demo on its storepage, and it has a separate page for its demo Outpath: First Journey
https://store.steampowered.com/app/2237970/Outpath/
https://store.steampowered.com/app/2237980/Outpath_First_Journey/
they are the exact same file, and share save data
but they dont cross over like this

Outpath: First Journey *does* show up in my Library

**Comment by mutsurhea** [comment]
one of the devs [replied](https://twitter.com/guilemus_/status/1735022778800128319)

> how come there was a demo, but its now disabled?
> the steam api can still see it and that can be used to play it still

> The demo was in a differrent stage from the final game,
> there are now new mechanichs and the game overall plays diferently
> so it would not be fair to present it as a representation of the game.

i think this is a likely explanation for all the hidden demos
might be worth hiding them,
or marking them in some way
*!hidden, does not reflect the final game!*
maybe its own toggle box

**Comment by cristan2** [comment]
Regarding Outpath - this was a common practice at one point for devs to present their demos as "prologues", basically a free chapter of a game listed as a separate application (with different ids and everything), it's not a normal demo per se. The practice has its pros and cons, but some devs still use it.

As for hidden demos, I hadn't considered this case. I'll try to see if Steam's API offers an option to differentiate between them.

**Comment by mutsurhea** [comment]
i think people would appreciate being able to order by

demo file size [failing that, game file size]
game positivity rating, like % of positive reviews
release date
price

 you dont need to use a 3rd party to find your SteamID
its in Account Details
https://store.steampowered.com/account/

**Comment by cristan2** [comment]
Appreciate the feedback, thanks!

**Comment by mutsurhea** [comment]
i wouldn't imagine you can / the api can do this, but it might
but it would be handy when working through your demos if you could mark-off what's been played
or if you could hide which demos you've played

can the api tell you if a user has played a demo?

in my Recently played i have all the demos
 but they're not in All Games [unless they made a separate database entry]
i think Recently is only last 2 week, does that mean data on user demo playtime is deleted after that?

maybe demo playtime is put as playtime for the whole game,
so if someone has playtime on a game on Wishlist, that means its demo playtime

[i myself am already using a spreadsheet to organise my stuff,
so i dont need this,
 however for other who aren't spreadsheeting lists of media they consume,
i think this'd be useful for them]

**Comment by mutsurhea** [comment]
i think twice ive had duplicate of entries
this was one
https://store.steampowered.com/app/1668690/Alina_of_the_Arena/

**Comment by cristan2** [comment]
I don't know yet if it's possible to show if a demo has been played. Funny, I do the same thing with my media, I use a spreadsheet for everything :)

I'll try to see if I can reproduce the duplicate entries. Thanks for telling me.

**Comment by mutsurhea** [comment]
FINISHED!

i vetted 234 demos
178 games were removed from my wishlist

couldn't have done it without out
tyvm

these were the demos i liked the most
> Outpath: First Journey
> REAVER
> Monomyth
> Tainted Grail: The Fall of Avalon
> Anger Foot

**Comment by cristan2** [comment]
I only heard of Anger Foot from your list, but Monomyth looks fun.

Glad you found the tool useful. Cheers!

**Comment by mutsurhea** [comment]
found something odd

MONSTER HUNTER RISE
https://store.steampowered.com/app/1446780/MONSTER_HUNTER_RISE/

on its page HAS a demo, in a green box
"Download Monster Hunter Rise: Sunbreak Demo"

this isn't showing in the app at all
[probably not alone]

also MHR: Sunbreak, at first glance, isnt the main game itself? its a DLC
https://store.steampowered.com/app/1880360/Monster_Hunter_Rise_Sunbreak/
so confused on the situation
 but maybe thats why the app misses it?

 but ALSO that demo has a separate page
https://store.steampowered.com/app/1836450/Monster_Hunter_Rise_Sunbreak_Demo/

probably not in scope
but it'd be handy if these could somehow be found by the app too
[the demos which are moved to their own page, or put in a free prequel or something]
 but i assume it'd not be easy so dont bother

**Comment by mutsurhea** [comment]
fyi steam Next Fest, demo event
https://store.steampowered.com/sale/nextfest

**Comment by mutsurhea** [comment]
i downloaded a bunch in the fest but not got to till now
not only are many demos now gone from page
but also gone if you installed it too

![alt text](https://i.imgur.com/mSHM25b.png)
![alt text](https://i.imgur.com/xgMBZc7.png)

 because

![alt text](https://i.imgur.com/mLHUbGf.png)

no servers no game

you can technically still load them

![](https://i.imgur.com/El547Y7.png)

 but
no servers,
 no game

i downloaded like 50 demos in the fest
only going through now
like 20 of them are down
so annoying

**Comment by targetbsp** [comment]
Is this site currently working?  For a few weeks now I've been getting 0 demo's back, and there definately are some!  It used to work great.
![image](https://github.com/user-attachments/assets/201150b1-483f-4115-8960-5a021689c331)

**Comment by mutsurhea** [comment]
i double checked
the list loads, 
the list loads x10 quicker than it used to
but the finding demo feature fails

![eVLjlTw](https://github.com/user-attachments/assets/0313b5eb-6228-43f5-a362-c983b3b20afa)

**Comment by cristan2** [comment]
Steam changed the API endpoint I was using (which was an unsupported way to read demos, by the way), but I haven't had the time to update the app. I'll try to fix it in time for the winter sale.

**Comment by mutsurhea** [comment]
im very glad to hear that
thx

**Comment by mutsurhea** [comment]
found another edge case
im not saying this is an error and need fixing
just documenting that the case exists

Hell Pie
game on steam, no demo https://store.steampowered.com/app/889910/Hell_Pie/
but
demo IS on gog https://www.gog.com/en/game/hell_pie_demo

**Comment by cristan2** [comment]
Just pushed an update with a quick fix, could you try it out? Thanks.

There will still be some gaps in the available games for now, but at least everything should work faster. And I'm working on a few more improvements for the next weeks.

One small improvement that is already available is the way the results page is returned, you can now bookmark the results for your own profile without having to search for it every time.

> Hell Pie
game on steam, no demo https://store.steampowered.com/app/889910/Hell_Pie/
but
demo IS on gog https://www.gog.com/en/game/hell_pie_demo

Nothing that can be done about this, I'm afraid, it's the developer's/publisher's choice, but thanks for spotting.

**Comment by targetbsp** [comment]
That seems to work now!  Nice one.  Thanks very much. :)

**Comment by mutsurhea** [comment]
thank you so much, thats a fab xmas present <3
and WOW it loaded INSTANTLY!!! wtf
before it took minutes and minutes and this was a blink
or was it pre-calculated as a return user?

**Comment by mutsurhea** [comment]
i only have 1 request

> If you don't know your SteamID, use a site such as Steam ID Finder or follow the instructions in the Steam FAQ.

there is an easier way, just hyperlink to here, it says your ID at the top under your username
https://store.steampowered.com/account/

![image](https://github.com/user-attachments/assets/24a8d8f5-554f-4c1b-9f8d-3f80e3bd16d2)

[this is how you navigate to this page]
![image](https://github.com/user-attachments/assets/579ed4ac-b5e5-4d55-a4bd-540988722b9f)

**Comment by cristan2** [comment]
> before it took minutes and minutes and this was a blink
or was it pre-calculated as a return user?

Yes, before the game info was retrieved on demand and that's why it took longer, but now it's stored locally, so the only request to steam is for the wishlist.

> there is an easier way, just hyperlink to here, it says your ID at the top under your username

Thanks, forgot about that. But I have plans to update the search box to accept more identifiers, including steam user, to avoid having to dig for the steam id in different places.

**Comment by mutsurhea** [comment]
oh nice

found another weird edge case
Dragon Age™: The Veilguard
https://store.steampowered.com/app/1845910/Dragon_Age_The_Veilguard/

in its "demo" is not a demo
![image](https://github.com/user-attachments/assets/0f95d3a7-a719-4b5e-a476-5f2bea52d781)
steam calls it a demo when viewed in this
![image](https://github.com/user-attachments/assets/2d3c7546-5e54-441b-91b7-addac9725de5)
> Download Dragon Age™: The Veilguard Character Creator

Dragon's Dogma 2 has it too
https://store.steampowered.com/app/2054970/Dragons_Dogma_2/
> Download Dragon's Dogma 2 Character Creator & Storage
